src/huggingface_hub/_commit_api.py

```diff
@@ -424,3 +425,6 @@ def prepare_commit_payload(
         ],
         "deletedFiles": [{"path": del_op.path_in_repo} for del_op in deletions],
     }
+    if parent_commit is not None:
+        payload["parentCommit"] = parent_commit
```

Comment:
We need `parentCommit` to be at the beginning of the payload, at least before `"files"` (see https://github.com/huggingface/moon-landing/pull/3528)